code_1.py

```diff
@@ -30,14 +32,56 @@
         for x, y in pressed:
             button = (x + y * 8) + 1
             pressed_buttons.append(button)
-            trellis.pixels[x, y] = (255, 0, 0)  # Set pressed buttons to red
         print("Pressed buttons:", pressed_buttons)
         gp.press_buttons(*pressed_buttons)
         for i in range(1, 17):
             if i not in pressed_buttons:
                 gp.release_buttons(i)
-                trellis.pixels[(i - 1) % 8, (i - 1) // 8] = (0, 0, 0)  # Set released buttons to black
     else:
         gp.release_all_buttons()
-        trellis.pixels.fill((0, 0, 0))  # Set all buttons to black
-    time.sleep(0.001)
\ No newline at end of file
+    if serial.connected:
+        while serial.in_waiting >= 12:
+            # Serial byte packed format: LLRRRGGGBBB\n
+            # LL: Led number
+            # RRR: Red intensity
+            # GGG: Green intensity
+            # BBB: Blue intensity
+            command = serial.readline()
+            if command is None:
+                break
+            if (len(command)) != 12:
+                print("Unexpected command length: ", len(command))
+                clearLeds(trellis)
+                serial.reset_input_buffer()
+                continue
+            commandString = str(command[:11], 'ascii')
+            print("Received command:", str(commandString))
+
+            if not commandString.isdigit():
+                print("Not a numeric command. Clearing!")
+                clearLeds(trellis)
+                continue
+
+            ledNumber = int(commandString[:2])
+            redValue = int(commandString[2:5])
+            greenValue = int(commandString[5:8])
+            blueValue = int(commandString[8:11])
+            if not inRange(ledNumber, 1, 32):
```

Comment:
are these really not 0 indexed?

Reply:
I did it this way to match the buttons, which start at 1 per the USB spec.